pkg/apis/core/validation/validation.go

```diff
@@ -5219,7 +5219,7 @@ func validateServiceExternalTrafficPolicy(service *core.Service) field.ErrorList
 	if !needsExternalTrafficPolicy(service) {
 		if service.Spec.ExternalTrafficPolicy != "" {
 			allErrs = append(allErrs, field.Invalid(fldPath.Child("externalTrafficPolicy"), service.Spec.ExternalTrafficPolicy,
-				"may only be set when `type` is 'NodePort' or 'LoadBalancer'"))
+				"may only be set when `type` is 'NodePort' or 'LoadBalancer', or when `type` is `ClusterIP` and `externalIPs` is set"))
```

Comment:
"may only be set for externally-reachable services"?

Reply:
updated